addons/web/static/src/js/core/dom.js

```diff
@@ -343,19 +343,38 @@ var dom = {
             // part, the button is disabled without any visual effect.
             $button.addClass('o_debounce_disabled');
             Promise.resolve(dom.DEBOUNCE && concurrency.delay(dom.DEBOUNCE)).then(function () {
-                $button.addClass('disabled').prop('disabled', true);
                 $button.removeClass('o_debounce_disabled');
-
-                return Promise.resolve(result).then(function () {
-                    $button.removeClass('disabled').prop('disabled', false);
-                }).guardedCatch(function () {
-                    $button.removeClass('disabled').prop('disabled', false);
-                });
+                dom.addButtonLoadingEffect($button[0], true);
+                const always = () => {
+                    dom.addButtonLoadingEffect($button[0]);
+                };
+                return Promise.resolve(result).then(always).guardedCatch(always);
             });
 
             return result;
         };
     },
+    /**
+     * Gives the given button a loading effect by disabling it and adding a `fa`
+     * spinner icon.
+     * The existing button `fa` icons will be hidden through css.
+     *
+     * @private
+     * @param {HTMLElement} btn - the button to disable/load
+     * @param {Boolean} [spin=false] - true to disable, false to enable
+     */
+    addButtonLoadingEffect: function (btn, spin = false) {
+        // TODO: Should we prevent this in edit mode? and allow for "Save" with
+        //       a force parameter such as ev.data.forceInEditMode
```

Comment:
Not sure what you mean, normally all event handler on button are disabled by the editor anyway isn't it?